File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.cpp
// Last updated: 7/28/2025, 12:30:14 PM
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int sum = 0;
        int n = nums.size();
        int maxSum =0;
        for(int i=0;i<n;i++){
            if(nums[i]==0) sum =0 ;
            sum = sum + nums[i];
            maxSum = max(sum , maxSum);
            

        }
        return maxSum ;
    }
};

File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.java
// Last updated: 7/28/2025, 12:30:20 PM
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int sum=0,maxi=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==1) 
            {
               sum=sum+1;
               maxi=Math.max(sum,maxi);
            }
            else sum=0;
        }
        return maxi;
    }
}